### PR TITLE
SAA-862:Offender merge event processing

### PR DESCRIPTION
**Pull request by timharrison-moj**
First PR for merge adds:
* A definition for the merge event content
* The event handler
* The structure of the processing required
* Logging of events received and the actions taken.
* Creation of a local audit event
* An event of interest - for display of the CoC screen.
* Tests for event processing

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  override fun handle(event: OffenderMergedEvent): Outcome {
+    log.info("MERGE: Handling offender merged event {}", event)
+    val newNumber = event.prisonerNumber()
+    val oldNumber = event.removedPrisonerNumber()
+
+    prisonerSearchApiClient.findByPrisonerNumber(newNumber)?.let { prisoner ->
+      log.info("MERGE: Search $newNumber = ${prisoner.firstName} ${prisoner.lastName} ${prisoner.prisonId} ${prisoner.status}")
+      prisoner.prisonId?.let { prisonId ->
+        if (rolloutPrisonRepository.prisonIsRolledOut(prisonId)) {
+          transactionHandler.newSpringTransaction {
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt:48]: Do you need to wrap the call to the `processMergeEvent` in a separate/new transaction?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt:48]: I think the transaction handler has that annotation - `@Transactional(propagation = REQUIRES_NEW)`

Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt:48]: Sure, but do you need a new independent transaction is my meaning?  There are no events being raised here.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/handlers/OffenderMergedEventHandler.kt:48]: Oh I see. Don't need it yet, but I will in the next PR.

**Pull request merged**